src/configPolling.ts

```diff
@@ -0,0 +1,144 @@
+import fetch from 'node-fetch';
+
+import ConfigService from './config';
+import {logger} from './logging';
+import {PollingConfig} from './types';
+import {validateConfig} from './validate';
+
+type StartOptions = {
+  /**
+   * Immediately trigger a polling tick when starting
+   */
+  immediate: boolean;
+};
+
+/**
+ * The ConfigPoller is used for two things:
+ *
+ *  1. During boot we poll on the bootInterval to load the configuration module.
+ *     Typically this will be a fairly quick poll to boot the service ASAP.
+ *
+ *  2. After it's booted, it will switch to the idleInterval. This is typically
```

Comment:
I'd argue we should also have an endpoint that triggers a fetch in a future PR so we can force a config reload on a known Sentry deploy.